bob/src/api/http.rs

```diff
@@ -469,8 +470,16 @@ async fn partition_by_id(
     let holders = group.holders();
     debug!("get pearl holders: OK");
     let pearls = holders.read().await;
-    let pearl = pearls.iter().find(|pearl| pearl.get_id() == partition_id);
+    let mut partitions = vec![];
```

Comment:
Seems like useless code. You already have ids in `pearls`. And you don't use extracted ids in `partitions` where you can't get id by `pearl.get_id()`